odbc/src/api/data.rs

```diff
@@ -775,12 +769,27 @@ pub unsafe fn format_bson_data(
         }
         _ => {}
     }
+
+    let uuid_repr = match &mongo_handle.as_connection() {
+        Some(conn) => match conn.mongo_connection.read() {
+            Ok(conn) => {
+                if conn.as_ref().is_some() {
+                    conn.as_ref().unwrap().uuid_repr
+                } else {
+                    None
+                }
+            }
+            Err(_) => None,
+        },
+        None => None,
+    };
+
```

Comment:
By storing it in the connection and reading it here we can support the desired uuid represenation per connection.